File: Golang/MissingNumber.go
func missingNumber(nums []int) int {

    ans := len(nums)

    for index, value := range nums {
        ans = ans ^ index ^ value
    }

    return ans
    
}

#2

func missingNumber(nums []int) int {
    res := len(nums)
    
    for i := 0; i < len(nums); i++ {
        res += i - nums[i]
    }
    return res
}

File: Python3/MissingNumber.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        rt = len(nums)
        for i in range(len(nums)):
            rt = rt ^ i ^ nums[i]

        return rt


#2

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        # use math
        total = int((1 + len(nums)) / 2 * len(nums))

        sum = 0

        for n in nums:
            sum += n

        return total - sum

#3

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        res = len(nums)

        for i in range(len(nums)):
            res += i - nums[i]
        return res
